### PR TITLE
CXP-628: decouple events api limit increment

### PR DESCRIPTION
**Pull request by pchasle**
Decouple the incrementation of the Events API rate limit from the main command handler.

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestFailedEvent.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Akeneo\Connectivity\Connection\Domain\Webhook\Event;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class EventsApiRequestFailedEvent
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestFailedEvent.php:9]: final

File: src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestFailedEvent.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestFailedEvent.php:2]: declare strict type

File: src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestSucceededEvent.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestSucceededEvent.php:2]: declare strict type

File: src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestSucceededEvent.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+namespace Akeneo\Connectivity\Connection\Domain\Webhook\Event;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class EventsApiRequestSucceededEvent
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Event/EventsApiRequestSucceededEvent.php:9]: final

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/EventsApiRequestsLimitIncrementSubscriberSpec.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+class EventsApiRequestsLimitIncrementSubscriberSpec extends ObjectBehavior
+{
+    public function let(EventsApiRequestCountRepository $repository): void
+    {
+        $this->beConstructedWith($repository);
+    }
+
+    public function it_is_initializable(): void
+    {
+        $this->shouldHaveType(EventsApiRequestsLimitIncrementSubscriber::class);
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/EventsApiRequestsLimitIncrementSubscriberSpec.php:28]: shold implement EventSubscriberInterface

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php [hunk 6]
```diff
@@ -53,7 +56,8 @@ public function it_is_initializable(): void
 
     public function it_sends_webhook_requests_in_bulk(
         SendApiEventRequestLogger $sendApiEventRequestLogger,
-        EventsApiRequestLogger $eventsApiRequestLogger
+        EventsApiRequestLogger $eventsApiRequestLogger,
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php:59]: I don't see any `$eventDispatcher->dispatch(EventsApiRequestFailedEvent)->shouldBeCalled()` in the spec. I think you missed to test something. Same about `EventsApiRequestSucceededEvent`.

**Review by willy-ahva — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php (hunk 6 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php:61]: fixed

**Review by tseho — APPROVED** [review]

**Pull request merged**